lib/ts/recipe/session/with-jwt/sessionClass.ts

```diff
@@ -55,8 +56,78 @@ export default class SessionClassWithJWT implements SessionContainerInterface {
+        const origSessionClaimPayloadJSON = JSON.stringify(this.getSessionClaimPayload());
+        const origAccessTokenPayloadJSON = JSON.stringify(this.getAccessTokenPayload());
+
+        let newSessionClaimPayload = this.getSessionClaimPayload();
+        let newAccessTokenPayload = this.getAccessTokenPayload();
+        for (const claim of claims) {
+            const value = await claim.fetch(this.getUserId(), userContext);
+            if (value !== undefined) {
+                newSessionClaimPayload = claim.addToPayload(this.getSessionClaimPayload(), value, userContext);
+            }
+            if (claim.updateAccessTokenPayload) {
+                newAccessTokenPayload = claim.updateAccessTokenPayload(
+                    this.getAccessTokenPayload(),
+                    value,
+                    userContext
+                );
+            }
+        }
+        const sessionClaimPayloadUpdate =
+            JSON.stringify(newSessionClaimPayload) !== origSessionClaimPayloadJSON ? newSessionClaimPayload : undefined;
+        const accessTokenPayloadUpdate =
+            JSON.stringify(newAccessTokenPayload) !== origAccessTokenPayloadJSON ? newAccessTokenPayload : undefined;
+        if (accessTokenPayloadUpdate !== undefined || sessionClaimPayloadUpdate !== undefined) {
+            await this.regenerateToken(accessTokenPayloadUpdate, sessionClaimPayloadUpdate, userContext);
+        }
+    };
+    checkClaimInToken = (claim: SessionClaim<any>, userContext?: any): Awaitable<boolean> => {
+        return claim.isValid(this.getSessionClaimPayload(), userContext);
+    };
+    addClaim = async <T>(claim: SessionClaim<T>, value: T, userContext?: any): Promise<void> => {
+        const newSessionClaimPayload = claim.addToPayload(this.getSessionClaimPayload(), value, userContext);
+
+        let newAccessTokenPayload;
+        if (claim.updateAccessTokenPayload) {
+            newAccessTokenPayload = claim.updateAccessTokenPayload(this.getAccessTokenPayload(), value, userContext);
+        }
+        await this.regenerateToken(newAccessTokenPayload, newSessionClaimPayload, userContext);
+    };
+    removeClaim = async <T>(claim: SessionClaim<T>, userContext?: any): Promise<void> => {
+        const newSessionClaimPayload = claim.removeFromPayload(this.getSessionClaimPayload(), userContext);
+
+        let newAccessTokenPayload;
+        if (claim.updateAccessTokenPayload) {
+            newAccessTokenPayload = claim.updateAccessTokenPayload(
+                this.getAccessTokenPayload(),
+                undefined,
+                userContext
+            );
+        }
+
+        await this.regenerateToken(newAccessTokenPayload, newSessionClaimPayload, userContext);
+    };
 
     updateAccessTokenPayload = async (newAccessTokenPayload: any, userContext?: any): Promise<void> => {
+        await this.originalSessionClass.regenerateToken(newAccessTokenPayload, undefined, userContext);
```

Comment:
This should call original implementation for `updateAccessTokenPayload`? (with `bind(this)`)?